src/components/assignment/js/assignment.js

```diff
@@ -0,0 +1,209 @@
+    autoAssignAll: function () {
+        this._showUpdating();
+        var maxUsers = +this.$('#max-users').val();
+        return this.worlds.autoAssignAll({ maxUsers: maxUsers })
+            .done(this._hideUpdating)
+            .fail(this._hideUpdating)
+            .then(function () {
+                this.worlds.joinUsers();
+                this.render();
+            }.bind(this));
+    },
+
+    usassignSelected: function (e) {
+        e.preventDefault();
+
+        var ids = this.$('tbody :checkbox:checked').map(function () {
+            return $(this).data('id');
+        });
+
+        var done = _.after(ids.length, function () {
+            this.worlds.fetch().then(function () {
+                this.worlds.joinUsers();
+                this._hideUpdating();
+                this.render();
+
+            }.bind(this));
+        }.bind(this));
+
+        this._showUpdating();
+        _.each(ids, function (userId) {
+            var user = this.users.getById(userId);
+            user.set('world', '');
+            user.set('role', '');
+            this.worlds.updateUser(user)
+                .done(done);
+        }, this);
+    },
+
+    render: function () {
+        this.$('table tbody').empty();
+        this.renderTable();
+        this.toggleControlls();
+    },
+
+    renderTable: function () {
+        var rows = [];
+        this.users.each(function (u) {
+            var view = new AssignemntRow({ model: u, worlds: this.worlds, project: this.project });
+            rows.push(view.render().el);
+        }, this);
+
+        this.$('table tbody').append(rows);
+    },
+
+
+    updateControls: function () {
+        this.updateControlsForSelection();
+        this.updateAutoAssignButton();
+        this.updateStatus();
+    },
+
+    updateStatus: function () {
+        var incolpleteWorlds = this.worlds.getIncompleteWorldsCount();
+        var unassignedUsers = this.users.getUnassignedUsersCount();
+        var totalWorlds = this.worlds.length();
+
+        var usersText = unassignedUsers ? unassignedUsers === 1 ? '1 user needs assignment.' : unassignedUsers + ' users need assignment.' : 'All users have been assigned.';
+        var worldsText = !totalWorlds ? 'No worlds have been created.' : !incolpleteWorlds ? 'All worlds are complete.' : incolpleteWorlds === 1 ? '1 incomplete world needs attention.' : incolpleteWorlds + ' incomplete worlds need attention.';
+
+        this.$('#users-status .text').text(usersText);
+        this.$('#worlds-status .text').text(worldsText);
+
+        if (unassignedUsers) {
+            this.$('#users-status').addClass('incomplete');
+        } else {
+            this.$('#users-status').removeClass('incomplete');
+        }
+
+        if (incolpleteWorlds || !totalWorlds) {
+            this.$('#worlds-status').addClass('incomplete');
+        } else {
+            this.$('#worlds-status').removeClass('incomplete');
+        }
+
+        this.$('.status-widget').css({ opacity: 1 });
```

Comment:
Can we control the opacity through classes? Can add transitions etc on it later then